ruby-challenges.rb

```diff
@@ -5,28 +5,73 @@
 
 beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
 
-letter_o = 'o'
-# Expected output: ['coffee', 'soda water']
-letter_t = 't'
-# Expected output: ['tea', 'water', 'soda water']
+# letter_o = 'o'
+# # Expected output: ['coffee', 'soda water']
+# letter_t = 't'
+# # Expected output: ['tea', 'water', 'soda water']
+
+
+def word_with_letters(givenArray, string)
+  givenArray.filter {|newArray| newArray.include? string}
+end
+p word_with_letters(beverages_array, letter_o)
+p word_with_letters(beverages_array, letter_t)
+#output= ["coffee", "soda water"]
+#output= ["tea", "water", "soda water"]
+
 
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
+
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
 
+# flatten takes in multiple arrays and turns them into one array (ruby get rid of nested arrays)
+# use .sort to sort the states alphabetically
+
+p us_states.values.flatten.sort 
```

Comment:
While this code will give you the expected results, it is not dynamic or reusable.  Wrapping this with a method will make this code block more dynamic.